actionview/lib/action_view/template/handlers/erb/erubi.rb

```diff
@@ -42,7 +42,7 @@ def add_text(text)
             end
           end
 
-          BLOCK_EXPR = /\s*((\s+|\))do|\{)(\s*\|[^|]*\|)?\s*\Z/
```

Comment:
@jhawthorn Could you please add/update the test case for this? which will help for future enhancement!